datastream-server/src/main/java/com/linkedin/datastream/server/Coordinator.java

```diff
@@ -689,7 +694,8 @@ private DatastreamTask getDatastreamTask(String taskName) {
 
     if (isDatastreamUpdate || !addedTasks.isEmpty() || !removedTasks.isEmpty()) {
       // Populate the event producers before calling the connector with the list of tasks.
-      addedTasks.stream().filter(t -> t.getEventProducer() == null).forEach(this::initializeTask);
+      addedTasks.stream().filter(t -> t.getEventProducer() == null).forEach(
+          task -> initializeTask(task, isDatastreamUpdate, retryAndSaveError));
       return submitAssignment(connectorType, assignment, isDatastreamUpdate, connector, removedTasks, retryAndSaveError);
```

Comment:
`submitAssignment` will get called with the assignment list containing the failedTasks. You can trim the list inside the `submitAssignment` method.

Reply:
Done